### PR TITLE
feat(ingest): update snowflake docs, add simple validations

### PR DESCRIPTION
**Pull request by mayurinehate**
- Remove starter recipe for legacy sources
- Simplify starter recipe for snowflake

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/docs/sources/snowflake/snowflake_recipe.yml [hunk 1]
```diff
@@ -1,11 +1,8 @@
 source:
   type: snowflake
   config:
-    # This option is recommended to be used for the first time to ingest all lineage
+    # This option is recommended to be used to ingest all lineage
```
Comment (mayurinehate) [inline metadata-ingestion/docs/sources/snowflake/snowflake_recipe.yml:4]: After patch lineage is supported for snowflake, this line can be updated to  original text - "This option is recommended to be used for the first time to ingest all lineage".
Currently, if this is removed, there is a possibility of overriding past lineage for same table.

File: metadata-ingestion/src/datahub/ingestion/source_config/sql/snowflake.py [hunk 2]
```diff
@@ -41,6 +41,8 @@
     "OAUTH_AUTHENTICATOR": OAUTH_AUTHENTICATOR,
 }
 
+SNOWFLAKE_HOST_SUFFIX = ".snowflakecomputing.com"
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/ingestion/source_config/sql/snowflake.py:44]: nice refactor!

File: metadata-ingestion/tests/unit/test_snowflake_beta_source.py [hunk 3]
```diff
@@ -91,6 +92,21 @@ def test_account_id_is_added_when_host_port_is_present():
...
+            {
+                "username": "user",
+                "password": "password",
+                "account_id": "acctname.snowflakecomputing.com",
+                "database_pattern": {"allow": {"^demo$"}},
+                "warehouse": "COMPUTE_WH",
+                "role": "sysadmin",
+            }
+        )
+    excinfo.value == f"Please remove trailing {SNOWFLAKE_HOST_SUFFIX} from `account_id`."
```
[the remark this answers is not in the document]
Comment (jjoyce0510) [reply metadata-ingestion/tests/unit/test_snowflake_beta_source.py:107]: +1

File: metadata-ingestion/src/datahub/ingestion/source_config/sql/snowflake.py [hunk 4]
```diff
@@ -161,16 +163,22 @@ def one_of_host_port_or_account_id_is_required(cls, values):
...
             if host_port is None:
                 raise ConfigurationError(
                     "One of account_id (recommended) or host_port (deprecated) is required"
                 )
             else:
                 values["account_id"] = host_port
+        else:
+            if account_id.endswith(SNOWFLAKE_HOST_SUFFIX):
+                raise ValueError(
+                    f"Please remove trailing {SNOWFLAKE_HOST_SUFFIX} from `account_id`."
```
[the remark this answers is not in the document]
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source_config/sql/snowflake.py:179]: got it. will change that.

**Pull request merged**